rio/test/RIO/VectorSpec.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+module RIO.VectorSpec where
+
+import Test.Hspec
+import Test.Hspec.QuickCheck (prop)
+import qualified Test.QuickCheck as QC
+import RIO
+import qualified RIO.Vector as V
+import qualified RIO.Vector.Partial as V'
+
+spec :: Spec
+spec =
+  describe "sliceMaybe" $ do
```

Comment:
I don't think it's necessary to _replace_ the unit tests with property tests, I was just hoping to include property tests as well.

Reply:
That was understood. However, I felt the property test was more to the point and completely specified the intended behavior of `sliceMaybe` (in terms of `slice`), thereby making the unit tests redundant.

Things became a bit complicated with the revealed bug of course.